File: 3622-check-divisibility-by-digit-sum-and-product/3622-check-divisibility-by-digit-sum-and-product.py
class Solution:
    def checkDivisibility(self, n: int) -> bool:
        sum=0
        product=1
        ans=n
        while n>0:
        
            rem=n%10
            sum+=rem
            product*=rem
            n=n//10
            print(n)
        
        return ans%(sum+product)==0
   

File: 3622-check-divisibility-by-digit-sum-and-product/3622-check-divisibility-by-digit-sum-and-product.java
class Solution {
    public boolean checkDivisibility(int n) 
    {
        int sum=0;
        int product=1;
        int ans=n;
        while(n>0)
        {
            int rem=n%10;
            sum+=rem;
            product *= rem;
            n=n/10;
        }
        return (ans % (sum+product) == 0) ? true : false;
        
    }
}